source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-icon/index.tsx

```diff
@@ -2,13 +2,15 @@ import * as React from 'react';
 import AndriodIcon from '@material-ui/icons/Android';
 import { ResourceObject } from 'jsonapi-typescript';
 import WebIcon from '@material-ui/icons/Web';
+import MissingIcon from '@material-ui/icons/ErrorOutline';
 
 import { ProductAttributes } from '@data/models/product';
 import { PRODUCTS_TYPE, attributesFor } from '@data';
 
 const iconMap = {
   android: () => <AndriodIcon />,
   html: () => <WebIcon />,
+  [undefined]: () => <MissingIcon />
```

Comment:
Great!

Reply:
probably, yeah.

Reply:
looking at the data again -- I don't actually think application type should be used here. We should use product type, as an application type can have products of different types. So, we'll need to wait until that's implemented.